### ISSUE TITLE
Error response from daemon: open /var/lib/docker/overlay/

### ISSUE THREAD

**Issue by monoflash** [docker/docker#22131]
**Output of `docker version`:**

```
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 19:36:04 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 19:36:04 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 3
Server Version: 1.11.0
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: null host bridge
Kernel Version: 4.5.0-3-default
Operating System: openSUSE Tumbleweed (20160411) (x86_64)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 2.938 GiB
Name: web
ID: WUSR:FPAV:KOS6:5TAC:RTDU:IY2Q:K5Y3:3I22:ZGFA:MRIG:A7BJ:SGA4
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. docker run --rm -it alpine echo hello
2. Result:
```
Unable to find image 'alpine:latest' locally
latest: Pulling from library/alpine
420890c9e918: Pull complete 
Digest: sha256:9cacb71397b640eca97488cf08582ae4e4068513101088e9f96c9814bfda95e0
Status: Downloaded newer image for alpine:latest
docker: Error response from daemon: open /var/lib/docker/overlay/90bc6a6664794c973c8c775ebfc696f9b8139e165fef8651064b8c3d2bf7a712-init/merged/.dockerenv: operation not permitted.
See 'docker run --help'.
```

**Comment by runcom** [comment]
Ping @cyphar seems to be happening on Tumbleweed

**Comment by matelukas** [comment]
Hey @monoflash 

I can see the same issue when using ext3/4 as backing file system.

Besides error you've mentioned I can see following warrning in dmesg when trying to run container.

`[ 1102.255035] EXT4-fs warning (device sda1): ext4_file_open:387: Inconsistent encryption contexts: 35281/65995
`

There is patch a patch for this issue mentioned here.

https://lkml.org/lkml/2016/3/22/31

There is no issue when using btrfs or xfs as backing fs.

Cheers :)

**Comment by cyphar** [comment]
@monoflash Where did you get this package from? Or did you build it from source?

**Comment by thaJeztah** [comment]
Is this a duplicate of https://github.com/docker/docker/issues/20950, and fixed in kernel 4.5.2?

**Comment by monoflash** [comment]
```
# uname -r -i
4.5.0-3-default x86_64
```

**Comment by monoflash** [comment]
v1.11.0 i download from https://github.com/docker/docker/releases link "Linux 64bit tgz:"
Not build from source.

**Comment by mlaventure** [comment]
@monoflash could you try by using a different backing fs as suggested by https://github.com/docker/docker/issues/22131#issuecomment-212033791 ?

Also, and maybe harder, try using a 4.5.2 kernel as suggested by @thaJeztah ?

**Comment by monoflash** [comment]
Problem solved.
Thanks.

**Comment by cyphar** [comment]
If it's a kernel issue, I can submit a bug report to get it backported into the Tumbleweed kernel. You'd have to wait for the next kernel update though.

I'd also recommend using the Docker packages we ship on openSUSE, because we apply certain patches and other things to make sure that Docker works properly on SUSE systems.

**Comment by rainbeam** [comment]
Same issue on Debian Stretch.

```
$ docker version
Client:
 Version:      1.8.3
 API version:  1.20
 Go version:   go1.5.1
 Git commit:   f4bf5c7
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.3
 API version:  1.20
 Go version:   go1.5.1
 Git commit:   f4bf5c7
 Built:        
 OS/Arch:      linux/amd64
```

```
$ docker info
Containers: 126
Images: 398
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.5.0-1-amd64
Operating System: Debian GNU/Linux stretch/sid (containerized)
CPUs: 2
Total Memory: 7.697 GiB
Name: builder
ID: [redacted]
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
```

Same reproduction steps (`docker run -it alpine echo hello`).

Problem started recently following a system upgrade (including the kernel).

**Comment by rainbeam** [comment]
Fixed on switching to kernel 4.5.0-2 from sid. I'd expect this to get pushed to stretch within a couple of weeks.

**Comment by cyphar** [comment]
The fixes have been backported to the openSUSE kernels.
